### PR TITLE
SYCL IJ matrix

### PR DESCRIPTION
**Pull request by waynemitchell**
This PR implements the IJ matrix functionality in sycl for use on Intel GPUs.

### CODE REVIEW

File: src/IJ_mv/IJMatrix_parcsr_device.c [hunk 1]
```diff
@@ -11,10 +11,11 @@
  *
  *****************************************************************************/
 
+#include "_hypre_onedpl.hpp"
```
Comment (rfalgout) [inline src/IJ_mv/IJMatrix_parcsr_device.c:14]: Can you remind me again why this is not integrated into `_hypre_utilities.hpp`

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

File: src/IJ_mv/IJMatrix_parcsr_device.c (hunk 1 above)
Comment (waynemitchell) [reply src/IJ_mv/IJMatrix_parcsr_device.c:14]: The oneDPL headers need to be included before any other C/C++ stl includes in order to work properly, and this was the cleanest way I found of doing this. Note that wherever `_hypre_onedpl.hpp` is included, it must always be the first include.

Comment (rfalgout) [reply src/IJ_mv/IJMatrix_parcsr_device.c:14]: Thanks!  Now I remember.

**Review by liruipeng — APPROVED** [review]
Looks good to me (from a high level. I trust @waynemitchell on the sycl details). Thanks!

**Comment by waynemitchell** [comment]
To make things easier to merge into master, I am merging this PR into PR 737 and closing this out. PR 737 will then contain all the new sycl functionality.

**Pull request closed without merging**